src/app/clusters/valve-configuration-and-control-server/valve-configuration-and-control-server.cpp

```diff
@@ -422,6 +389,44 @@ void UpdateAutoCloseTime(uint64_t time)
         }
     }
 }
+
+CHIP_ERROR GetRemainingDuration(EndpointId endpoint, DataModel::Nullable<uint32_t> & duration)
+{
+    uint16_t epIdx = emberAfGetClusterServerEndpointIndex(endpoint, ValveConfigurationAndControl::Id,
+                                                          MATTER_DM_VALVE_CONFIGURATION_AND_CONTROL_CLUSTER_SERVER_ENDPOINT_COUNT);
+    VerifyOrReturnValue(epIdx < kValveConfigurationAndControlDelegateTableSize, CHIP_ERROR_READ_FAILED);
+    duration = gRemainingDuration[epIdx].remainingDuration;
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR SetRemainingDuration(EndpointId endpoint, DataModel::Nullable<uint32_t> duration)
+{
+    uint16_t epIdx = emberAfGetClusterServerEndpointIndex(endpoint, ValveConfigurationAndControl::Id,
+                                                          MATTER_DM_VALVE_CONFIGURATION_AND_CONTROL_CLUSTER_SERVER_ENDPOINT_COUNT);
+    if (epIdx < kValveConfigurationAndControlDelegateTableSize)
+    {
+        gRemainingDuration[epIdx].endpoint          = endpoint;
+        gRemainingDuration[epIdx].remainingDuration = duration;
+        return CHIP_NO_ERROR;
```

Comment:
you are changing an attribute here. Why is MatterReportingAttributeChangeCallback not invoked here (but is on the set null version)?